### ISSUE TITLE
azure_rm_deployment: breaks due to version number change in azure sdk 2.0.0rc4 or greater

### ISSUE THREAD

**Issue by justenwalker** [ansible/ansible-modules-extras#2554]
##### ISSUE TYPE
- Bug Report
##### COMPONENT NAME

azure_rm_deployment
##### ANSIBLE VERSION

```
ansible 2.1.0.0
  config file =
  configured module search path = Default w/o overrides
```
##### OS / ENVIRONMENT

N/A
##### SUMMARY

the `azure_rm_deployment` module cannot be run due to a change in the version spec in azure python sdk version 2.0.0rc4 or greater. Installing a prior RC resolves the error (2.0.0rc3)

the task produces an error about the sdk version (example with 2.0.0rc5):

```
Expecting azure.mgmt.compute.__version__ to be >= 2016-03-30. Found version 0.30.0rc5 Do you have Azure >= 2.0.0rc2 installed?
```
##### STEPS TO REPRODUCE
- Install azure python sdk: `pip install azure==2.0.0rc4`
- Create a play using `ansible_rm_deployment`
- Run the play
##### EXPECTED RESULTS

Deployment using azure resource templates is made to azure
##### ACTUAL RESULTS

```
fatal: [127.0.0.1]: FAILED! => {"changed": false, "failed": true, "msg": "Expecting azure.mgmt.compute.__version__ to be >= 2016-03-30. Found version 0.30.0rc5 Do you have Azure >= 2.0.0rc2 installed?"}
```

**Comment by justenwalker** [comment]
Upon further investigation, it seems that the main ansible repository fixed this [about a week ago](https://github.com/ansible/ansible/blob/54db1df24499ca99f44c61b24bef59176ccdfa21/lib/ansible/module_utils/azure_rm_common.py), but obviously that has not made it into the current GA release of Ansible yet.  For those looking for a work-around, install the azure sdk with: `pip install azure==2.0.0rc2`

Since this affects `lib/ansible/module_utils/azure_rm_common.py` this means it would be a problem in _all_ azure cloud modules.

**Comment by nitzmahone** [comment]
This fix has also been backported to stable-2.1, and is available in 2.1.1RC2. Closing.
